### PR TITLE
Thrust update 1.9.4-1.9.8

### PR DESCRIPTION
**Pull request by AJcodes**
Thrust update 1.9.4-1.9.8:
- Bug fixes from version 1.9.4 to 1.9.8

### CODE REVIEW

**Comment by AJcodes** [comment]
@saadrahim @doctorcolinsmith Turns out there was a simpler solution that allows your CI to pass; they updated the git submodule url to a local path. I've pushed a fix so the url points to the repository

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "cub"]
```
Comment (saadrahim) [inline .gitmodules:1]: @doctorcolinsmith @stanleytsang-amd Is this acceptable?

Comment (doctorcolinsmith) [reply .gitmodules:1]: Should the submodule point to our hipCUB instead of NVIDIA's cub?

Comment (AJcodes) [reply .gitmodules:1]: I've checked with Istvan and our cmake script we've developed for rocThrust does not directly support the CUDA backend (cmake will throw an error message if nvcc is selected). The decision for this was made at the initial stages of development, where it was agreed that users looking to use CUDA could revert to the official repo (especially given the official repo was versions ahead).

I'm not sure what your CI does with cub after cloning it (the solution was based on your CI), but perhaps after this is merged, we could discuss the best way around this.

Comment (saadrahim) [reply .gitmodules:1]: Makes sense to me. CUDA compatibility can be a separate discussion.

**Pull request merged**